src/Integration.UnitTests/TeamExplorer/ConnectSectionControllerTests.cs

```diff
@@ -347,72 +325,25 @@ public void ConnectSectionController_SetProjectsUIThread()
-
-            var serverA = new ServerViewModel(new ConnectionInformation(new Uri("http://sonarqube-A")));
-            serverA.SetProjects(new[]
-            {
-                new ProjectInformation { Key = "A_p1", Name = "A_Project1" },
-                new ProjectInformation { Key = "A_p2", Name = "A_Project2" }
-            });
-            IList<ProjectViewModel> projectsA = serverA.Projects.ToList();
-            projectsA[0].IsBound = true;
-
-            var serverB = new ServerViewModel(new ConnectionInformation(new Uri("http://sonarqube-B")));
-            serverB.SetProjects(new[]
-            {
-                new ProjectInformation { Key = "B_p1", Name = "B_Project1" },
-                new ProjectInformation { Key = "B_p2", Name = "B_Project2" }
-
-            });
-            IList<ProjectViewModel> projectsB = serverB.Projects.ToList();
-            projectsB[0].IsBound = true;
-
-            testSubject.ConnectedServers.Add(serverA);
-            testSubject.ConnectedServers.Add(serverB);
-            testSubject.BoundProjects.Add(projectsA[0]);
-            testSubject.BoundProjects.Add(projectsB[0]);
-
-            serverA.ShowAllProjects = false;
-            serverB.ShowAllProjects = false;
-            testSubject.Notification.ShowNotificationError("message", NotificationIds.FailedToFindBoundProjectKeyId, null);
-
-            // Act
-            testSubject.ClearBoundProjects(serverA);
-
-            // Verify
-            Assert.AreSame(projectsB[0], testSubject.BoundProjects.Single(), "Expected only server B bound project");
-            Assert.IsTrue(projectsB[0].IsBound, "Expected server B bound project to remain bound");
-            Assert.IsFalse(projectsB[1].IsBound, "Expected server B unbound project remain unbound");
-            Assert.IsFalse(serverB.ShowAllProjects, "Server B ShowAllProjects should remain false");
-
-            Assert.IsFalse(projectsA[0].IsBound, "Expected server A bound project to now be set as unbound");
-            Assert.IsFalse(projectsA[1].IsBound, "Expected server A unbound project remain unbound");
-            Assert.IsTrue(serverA.ShowAllProjects, "Server A ShowAllProjects should be true after clearing it's bound projects");
-            testSubject.NotificationOverride.AssertNoNotification(NotificationIds.FailedToFindBoundProjectKeyId);
-        }
+            var testSubject = this.CreateTestSubject();
+            testSubject.NotificationAccessor.ShowNotificationError("message", NotificationIds.FailedToFindBoundProjectKeyId, null);
+            testSubject.State.ConnectedServers.Add(new ServerViewModel(new ConnectionInformation(new Uri("http://zzz1"))));
+            testSubject.State.ConnectedServers.Add(new ServerViewModel(new ConnectionInformation(new Uri("http://zzz2"))));
+            testSubject.State.ConnectedServers.ToList().ForEach(s => s.Projects.Add(new ProjectViewModel(s, new ProjectInformation()) { IsBound = true }));
```

Comment:
No need to set the `ProjectViewModel`s as bound, since the call to `SetBoundProject` below will modify these properties anyway. Also this state should never occur in the product (more than 1 project set as bound at the same time).